packages/flutter/lib/src/material/drop_down.dart

```diff
@@ -377,6 +414,21 @@ class _DropDownButtonState<T> extends State<DropDownButton<T>> {
       _updateSelectedIndex();
   }
 
+  @override
+  void dispose() {
+    if (_currentRoute != null) {
+      assert(_currentRoute.isActive);
+      Navigator.openTransaction(context, (NavigatorTransaction transaction) {
+        transaction.popUntil(_currentRoute);
+        assert(_currentRoute.isCurrent);
+        transaction.pop();
+      });
```

Comment:
This is a bit scary, but I'm not sure what else to recommend.  Imagine a full-screen dialog gets placed over the menu.  This widget will be behind an opaque route and therefore disposed.  Now this code will unwind the history until it respawns.